packages/amesos2/src/Amesos2_MUMPS_def.hpp

```diff
@@ -496,12 +502,25 @@ namespace Amesos2
   void
   MUMPS<Matrix,Vector>::MUMPS_ERROR()const
   {
-    if(mumps_par.info[0] < 0)
-      {
-        TEUCHOS_TEST_FOR_EXCEPTION(false,
+    bool Wrong = ((mumps_par.info[0] != 0) || (mumps_par.infog[0] != 0)) && (this->rank_ < this->nprocs_);
+    if(Wrong){
+      if (this->rank_==0) {
+        std::cerr << "Amesos_Mumps : ERROR" << std::endl;
+        std::cerr << "Amesos_Mumps : INFOG(1) = " << mumps_par.infog[0] << std::endl;
+        std::cerr << "Amesos_Mumps : INFOG(2) = " << mumps_par.infog[1] << std::endl;
+      }
+      if (mumps_par.info[0] != 0  && Wrong) {
+        std::cerr << "Amesos_Mumps : On process " << this->matrixA_->getComm()->getRank()
+        << ", INFO(1) = " << mumps_par.info[0] << std::endl;
+        std::cerr << "Amesos_Mumps : On process " << this->matrixA_->getComm()->getRank()
+        << ", INFO(2) = " << mumps_par.info[1] << std::endl;
+      }
+      if (this->rank_==0) {
```

Comment:
Amesos2 should throw on all processes, if it expects all processes to know about the error.  Otherwise, some processes may continue on, which could result in a hang.  Do you consider this error recoverable?

Reply:
You are right, the error is now broadcasted so every rank can throw.